libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/barrier.h

```diff
@@ -857,4 +857,180 @@ async_contract_fulfillment memcpy_async(void * __destination, void const * __sou
+    : "memory");
+}
+
+
+// https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#data-movement-and-conversion-instructions-cp-async-bulk
+_LIBCUDACXX_DEVICE
+void cp_async_bulk_shared_to_global(void *dest, const void * src, uint32_t size)
+{
+  asm volatile(
+    "cp.async.bulk.global.shared::cta.bulk_group [%0], [%1], %2;\n"
+    :
+    : "l"(static_cast<uint64_t>(__cvta_generic_to_global(dest))),
+      "r"(static_cast<uint32_t>(__cvta_generic_to_shared(src))),
+      "r"(size)
+    : "memory");
+}
+
+// https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#data-movement-and-conversion-instructions-cp-async-bulk-tensor
+_LIBCUDACXX_DEVICE
+void cp_async_bulk_tensor_1d_global_to_shared(
+    void *dest, const void *tensor_map , int c0, cuda::barrier<cuda::thread_scope_block> &bar)
+{
+  asm volatile(
+    "cp.async.bulk.tensor.2d.shared::cluster.global.tile.mbarrier::complete_tx::bytes "
+    "[%0], [%1, {%2}], [%3];\n"
+    :
+    : "r"(static_cast<uint32_t>(__cvta_generic_to_shared(dest))),
+      "l"(tensor_map),
+      "r"(c0),
+      "r"(static_cast<uint32_t>(__cvta_generic_to_shared(cuda::device::barrier_native_handle(bar))))
+    : "memory");
+}
+
+// https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#data-movement-and-conversion-instructions-cp-async-bulk-tensor
+_LIBCUDACXX_DEVICE
+void cp_async_bulk_tensor_2d_global_to_shared(
+    void *dest, const void *tensor_map , int c0, int c1, cuda::barrier<cuda::thread_scope_block> &bar)
+{
+  asm volatile(
+    "cp.async.bulk.tensor.2d.shared::cluster.global.tile.mbarrier::complete_tx::bytes "
+    "[%0], [%1, {%2, %3}], [%4];\n"
+    :
+    : "r"(static_cast<uint32_t>(__cvta_generic_to_shared(dest))),
+      "l"(tensor_map),
+      "r"(c0),
+      "r"(c1),
+      "r"(static_cast<uint32_t>(__cvta_generic_to_shared(cuda::device::barrier_native_handle(bar))))
+    : "memory");
+}
+
+// https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#data-movement-and-conversion-instructions-cp-async-bulk-tensor
+_LIBCUDACXX_DEVICE
+void cp_async_bulk_tensor_3d_global_to_shared(
+    void *dest, const void *tensor_map, int c0, int c1, int c2, cuda::barrier<cuda::thread_scope_block> &bar)
+{
+  asm volatile(
+    "cp.async.bulk.tensor.2d.shared::cluster.global.tile.mbarrier::complete_tx::bytes "
+    "[%0], [%1, {%2, %3, %4}], [%5];\n"
+    :
+    : "r"(static_cast<uint32_t>(__cvta_generic_to_shared(dest))),
+      "l"(tensor_map),
+      "r"(c0),
+      "r"(c1),
+      "r"(c2),
+      "r"(static_cast<uint32_t>(__cvta_generic_to_shared(cuda::device::barrier_native_handle(bar))))
+    : "memory");
+}
+
+// https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#data-movement-and-conversion-instructions-cp-async-bulk-tensor
+_LIBCUDACXX_DEVICE
+void cp_async_bulk_tensor_4d_global_to_shared(
+    void *dest, const void *tensor_map , int c0, int c1, int c2, int c3, cuda::barrier<cuda::thread_scope_block> &bar)
```

Comment:
Do this for all the references to the `cuda::` namespace. 
```suggestion
    void *dest, const void *tensor_map , int c0, int c1, int c2, int c3, ::cuda::barrier<::cuda::thread_scope_block> &bar)
```